### ISSUE TITLE
To Do & Done Week of Dec. 8th

### ISSUE THREAD

**Issue by alyciakb** [UBC-MDS/DSCI_522_Alberta-Oil-Spills#40]
### Decision Tree

- Printing straight to png, no longer prints to PDF. I updated the makefile and code comments to reflect the change.
- Changed output name from `oil_spills_model.png` to `spills_tree_model.png`
- Rotated decision tree to print horizontally and be easier to read in the final report
- Removed gini scores from decision tree to have less text
- Tried and failed to remove 'values' and 'samples' from it (I think it's fine how it is)

### Code of Conduct

-  added a code of conduct to include with our project, called CONDUCT.md. **Please review it. **

### Dockerfile

- wrote Dockerfile and tested locally

### Updates to existing files

- updated code in 4_model_viz to print decision tree horizontally, remove gini scores from tree visual, and print directly to PDF
- updated README to include the versions of the dependencies and previous repo releases
- updated README, makefile to remove code line that converts PDF to png (no longer needed)
- removed PDF version of decision tree

## STILL TO DO

- [x] create dependency diagram of makefile
- [x] Add unit test for `save_and_show_decision_tree` (Milestone 2, issue 3)
- [x] Discuss decision tree in more detail in final report (Milestone 2, issue 2)
- [x] Add scores to the features predictive rankings, so that people can better understand their ranking and their score as an indicator (Milestone 2, issue 4 and peer feedback)
- [x] Rearrange portions of the final report, add graphics to appendix, add depth_score.png to appendix (Milestone 2 comment, and peer feedback from last week)
- [x] Create auto build in Docker Hub (Alycia is currently doing this)
- [x] update README to include docker instructions

**Comment by huijuechen** [comment]
**DONE BELOW**:
- Add unit test for save_and_show_decision_tree (Milestone 2, issue 3)
- Discuss decision tree in more detail in final report (Milestone 2, issue 2)
- Add scores to the features predictive rankings, so that people can better understand their ranking and their score as an indicator (Milestone 2, issue 4 and peer feedback)

**ALSO**
Added depth_compare.png to the report, and briefly commented on it, related to our choice of max_depth in the actual tree (Milestone 2, issue 5)

**Comment by alyciakb** [comment]
- Reviewed the tree interpretation in the final report and added a couple sentences about the final predictions made by decision tree.
- Reviewed the gini interpretations in the final report. I rounded the gini scores to 5 decimals in the table. 
- The conduct file looks good! 
- Completed some grammatical and aesthetic edits to the README (editing what I did yesterday)
- Added `import os` to 4_model_viz.py because the makefile was throwing an error
